packages/react-utilities/src/compose/types.ts

```diff
@@ -113,6 +113,34 @@ export type ComponentState<Shorthands extends ObjectShorthandPropsRecord> = {
   };
 } & Shorthands;
 
+/**
+ * This is part of a hack to infer the element type from a native element *props* type.
+ * The only place the original element is found in a native props type (at least that's workable
+ * for inference) is in the event handlers, so some of the helper types use this event handler
+ * name to infer the original element type.
+ *
+ * Notes:
+ * - Using an extremely obscure event handler reduces the likelihood that its signature will be
+ *   modified in any component's props.
+ * - Inferring based on a single prop name instead of a larger type like `DOMAttributes<T>` should be
+ *   less expensive for typescript to evaluate and is less likely to result in type expansion in .d.ts.
+ */
+type SomeEventName = 'onLostPointerCaptureCapture';
+
+/**
+ * Return type for `React.forwardRef`, including inference of the proper typing for the ref.
+ */
+export type ForwardRefComponent<Props> = SomeEventName extends keyof Props
+  ? Required<Props>[SomeEventName] extends React.PointerEventHandler<infer Element>
+    ? React.ForwardRefExoticComponent<Props & React.RefAttributes<Element>>
```

Comment:
I believe `Props` should be updated like this unless we can always ensure there won't be any `ref` passed in.
```suggestion
    ? React.ForwardRefExoticComponent<React.PropsWithoutRef<Props> & React.RefAttributes<Element>>
```

Reply:
I was trying to avoid `PropsWithoutRef` because IIRC the odd way it's defined (either `React.PropsWithoutRef` or the variant in this file) increased the likelihood of type explosion. If components using this type follow our current patterns, a ref shouldn't be included, though of course there's no guarantee of that.